internal/oci/runtime_oci.go

```diff
@@ -824,19 +824,21 @@ func (r *runtimeOCI) StopContainer(ctx context.Context, c *Container, timeout in
 	ctx, span := log.StartSpan(ctx)
 	defer span.End()
 
+	if c.State().Status == ContainerStatePaused {
```

Comment:
this suffers the same way as the existing code, we need to move this check after we take the `opLock` in `StopLoopForContainer`